scripts/PlayoutControl_CLI.py

```diff
@@ -0,0 +1,246 @@
+# ./PlayoutControl_CLI.py -c playlist_track_moveup -v 1
+# ./PlayoutControl_CLI.py -c playlist_track_movedown -v 1
+# ./PlayoutControl_CLI.py -c playlist_track_remove -v 1
+# ./PlayoutControl_CLI.py -c playlist_clear
+# ./PlayoutControl_CLI.py -c playlist_loop -v playlist
+# ./PlayoutControl_CLI.py -c playlist_loop -v track
+# ./PlayoutControl_CLI.py -c playlist_loop -v off
+# ./PlayoutControl_CLI.py -c playlist_load_play -d "ZZZ/SubMaster/bbb-AudioFormatsTest" --playlistname "ZZZ % SubMaster % bbb-AudioFormatsTest"
+
+# ./PlayoutControl_CLI.py -c volume_up
+# ./PlayoutControl_CLI.py -c volume_down
+# ./PlayoutControl_CLI.py -c volume_get
+# ./PlayoutControl_CLI.py -c volume_set -v 80
+# ./PlayoutControl_CLI.py -c volume_system_set
+# ./PlayoutControl_CLI.py -c playout_mute_status
+# ./PlayoutControl_CLI.py -c playout_mute_toggle
+
+# ./PlayoutControl_CLI.py -c sys_volume_max_set -v 90
+# ./PlayoutControl_CLI.py -c sys_volume_max_get
+# ./PlayoutControl_CLI.py -c sys_volume_min_set -v 20
+# ./PlayoutControl_CLI.py -c sys_volume_min_get
+
+# ./PlayoutControl_CLI.py -c sys_config_value_set -k "hello" -v "world"
+# ./PlayoutControl_CLI.py -c sys_config_value_get -k "CMDNEXT"
+
+# ./PlayoutControl_CLI.py -c sys_volume_system_get
+# ./PlayoutControl_CLI.py -c sys_volume_system_set -v 30
+
+# ./PlayoutControl_CLI.py -c sys_volume_step_get
+# ./PlayoutControl_CLI.py -c sys_volume_step_set -v 3
+
+# ./PlayoutControl_CLI.py -c sys_idle_shutdown_get
+# ./PlayoutControl_CLI.py -c sys_idle_shutdown_set -v 20
+
+if(args_main['value']):
+    logger.debug('Convert --value "%s" argument str int or float type.' % (args_main['value']))
+    logger.debug('Current type: %s' % (type(args_main['value'])))
+    if(is_int(args_main['value'])):
+        args_main['value'] = int(args_main['value'])
+    else:
+        if(is_float(args_main['value'])):
+            args_main['value'] = float(args_main['value'])
+    logger.debug('New type: %s' % (type(args_main['value'])))
+
+if(args_main['key']):
+    logger.debug('Convert --key "%s" argument str int or float type.' % (args_main['value']))
+    logger.debug('Current type: %s' % (type(args_main['key'])))
+    if(is_int(args_main['key'])):
+        args_main['key'] = int(args_main['key'])
+    else:
+        if(is_float(args_main['key'])):
+            args_main['key'] = float(args_main['key'])
+    logger.debug('New type: %s' % (type(args_main['key'])))
+
+if(args_main['command']):
+    # COMMAND given
+    logger.debug('Try command: ' + args_main['command'])
+    if(args_main['command'].startswith("switch_")):
+        # SWITCH attempt
+        args_func = {}
+        args_func['command'] = args_main['command']
+        args_func['value'] = args_main['value']
+        args_func['dirpath'] = args_main['dirpath']
+        playProcess.sys_switches(args_func) 
+
+    elif(args_main['command'] == "playout_position_save"):
+        playProcess.playout_position_save()
+
+    elif(args_main['command'] == "playout_resume_play"):
```

Comment:
Also, have you considered to make a module with functions and call them by their name? This might allow you to have this in separated files (or at least individual (testable) methods).

https://stackoverflow.com/questions/3061/calling-a-function-of-a-module-by-using-its-name-a-string

@arne123 does the same on future3/develop in the [RPC Client](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/src/jukebox/jukebox/rpc/Server.py#L33-L35)

Reply:
You are right. at least the repetivie code like this would shrink into 1 line.

```python
elif(args_main['command'] == "playout_position_save"):
    playProcess.playout_position_save()
```